include/zephyr/drivers/sensor.h

```diff
@@ -50,6 +63,8 @@ struct sensor_value {
 	int32_t val1;
 	/** Fractional part of the value (in one-millionth parts). */
 	int32_t val2;
+	/** buffer part of the value*/
+	struct sensor_value_ex ex;
```

Comment:
It is important for larger chunks of data exported or imported to the driver and introducing ex is inevitable, the footprint is just 8 bytes in a 32-bit machine or shall be made as a confidential compilation.

Reply:
@MaureenHelm ping

Reply:
Unfortunately I agree with Maureen in this case. I don't think this change can be accepted.

Could the sensor_value two words be used to store the buf pointer and length instead?

Something like...

``` c
struct sensor_value fp_data = {
   .val1 = (int32_t)(buf),
   .val2 = (int32_t)(buf_len),
}
```

I'd note though that passing buffers to/from user mode threads will not work without special care to share the memory from kernel space to user space. So that would be a caveat for this particular driver and use case.

Reply:
I would recommend using the async API instead.

Reply:
There's an attribute that this seems to be used for